tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,946 @@
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+
+def update_progress_bar(total, progress):
+    """
+    Displays and updates a console progress bar.
+    """
+    barLength, status = 20, ""
+    progress = float(progress) / float(total)
+    if progress >= 1.:
+        progress, status = 1, "\r\n"
+
+    # Number of blocks to display. Used to visualize progress.
+    block = int(round(barLength * progress))
+    text = "\r[{}] {:.0f}% {}".format(
+        "#" * block + "-" * (barLength - block), round(progress * 100, 0),
+        status)
+
+    # Send the progress to stdout.
+    sys.stdout.write(text)
+
+    # Send the buffered text to stdout!
+    sys.stdout.flush()
+
+
+def filename_ends_with_extension(filename, extensions):
+    """Helper method to see if filename ends with certain extension"""
+    return reduce(lambda result, ext: filename.endswith("." + ext) or result, extensions, False)
+
+
+def inside_included_directories(dirpath, rootpath, include_dirs):
+    """Helper method to see if filename within included directories"""
+    return reduce(lambda result, included_directory: re.match(r'(%s)\b' % os.path.join(rootpath, included_directory), dirpath) or result, include_dirs, None)
+
+
+def walk_over_directory(rootpath, extensions, show_detailed=False, include_dirs=None):
+    """
+    Recursively walk over directory and call preprocessor on selected files.
+
+    Arguments)
+        extensions - A plist of file extensions ['cu', 'cuh', ..]
+
+        include_dirs - Directories under the rootpath that should be included in the walk.
+
+        show_detailed - Show a detailed summary of the transpilation process.
+    """
+
+
+    # Default argument for excluded directories.
+    if include_dirs is None:
+        include_dirs = []
+
+    # Compute the total number of files to be traversed.
+    total_files = 0
+    for (dirpath, _dirnames, filenames) in os.walk(rootpath):
+        if inside_included_directories(dirpath, rootpath, include_dirs):
+            for filename in filenames:
+                total_files += filename_ends_with_extension(filename, extensions)
+
+    current_file = 0
+
+    # Preprocessing statistics.
+    stats = {"unsupported_calls": [], "kernel_launches": []}
+
+    # Begin traversing the files.
+    for (dirpath, _dirnames, filenames) in os.walk(rootpath, topdown=True):
+        # Check if file ends with a valid extensions
+        if not inside_included_directories(dirpath, rootpath, include_dirs):
+            continue
+
+        for filename in filenames:
+            if filename_ends_with_extension(filename, extensions):
+                # Construct the file's full path
+                filepath = os.sep.join([dirpath, filename])
+
+                # Execute the preprocessor on the specified file.
+                preprocessor(filepath, stats)
+
+                # Update the progress
+                print(os.path.join(dirpath, filename))
+                update_progress_bar(total_files, current_file)
```

Comment:
You don't have to fix this now, but for future reference, whenever you add some sort of progress bar functionality, always include an option to disable it. Otherwise, these progress bars clog up CI logs without adding much of value.

Reply:
agreed.

Reply:
This should be fixed now. A simple addition to the command line invocation for hipify-python is needed.